### PR TITLE
feat: resolve-all functionality

### PR DESCRIPTION
**Pull request by toddbaert**
Closes: https://github.com/open-feature/schemas/issues/63

### CODE REVIEW

File: protobuf/schema/v1/schema.proto [hunk 1]
```diff
@@ -10,6 +10,39 @@ option php_namespace = "OpenFeature\\Providers\\Flagd\\Schema\\Grpc";
...
+    string variant = 2;
+
+    // The response value of the boolean flag evaluation, will be unset in the case of error.
+    oneof value {
+        bool                    boolValue = 3;
+        string                  stringValue = 4;
+        int64                   intValue = 5;
+        double                  doubleValue = 6;
+        google.protobuf.Struct  objectValue = 7;
+    }
```
Comment (toddbaert) [inline protobuf/schema/v1/schema.proto:43]: I think [oneof](https://developers.google.com/protocol-buffers/docs/proto3#oneof) is a good solution here. In many languages it generates some convenience functions so we can  easily check what value is set.

File: protobuf/schema/v1/schema.proto [hunk 2]
```diff
@@ -10,6 +10,39 @@ option php_namespace = "OpenFeature\\Providers\\Flagd\\Schema\\Grpc";
...
+
+    // Object structure describing the EvaluationContext used in the flag evaluation, see https://docs.openfeature.dev/docs/reference/concepts/evaluation-context
+    google.protobuf.Struct context = 1;
+}
+
+// Response body for bulk flag evaluation, used by the ResolveAll rpc.
+message ResolveAllResponse {
+
+    // Object structure describing the evaluated flags for the provided context.
+    map<string, AnyFlag> flags = 1;
```
Comment (toddbaert) [inline protobuf/schema/v1/schema.proto:24]: Is `flags` good enough? Or should we call it `evaluatedFlags` or `evaluated`?

File: protobuf/schema/v1/schema.proto [hunk 3]
```diff
@@ -10,6 +10,39 @@ option php_namespace = "OpenFeature\\Providers\\Flagd\\Schema\\Grpc";
...
+
+    // The reason for the given return value, see https://docs.openfeature.dev/docs/specification/types#resolution-details
+    string reason = 1;
+
+    // The variant name of the returned flag value.
+    string variant = 2;
+
+    // The response value of the boolean flag evaluation, will be unset in the case of error.
+    oneof value {
+        bool                    boolValue = 3;
```
Comment (james-milligan) [inline protobuf/schema/v1/schema.proto:38]: linting errors here as `boolValue` should instead be lower snake cased `bool_value`, same with the other names

File: protobuf/schema/v1/schema.proto (hunk 2 above)
Comment (james-milligan) [reply protobuf/schema/v1/schema.proto:24]: Im happy with `flags`

File: protobuf/schema/v1/schema.proto (hunk 3 above)
Comment (toddbaert) [reply protobuf/schema/v1/schema.proto]: Good call, resolved.

**Comment by james-milligan** [comment]
Looks good to me - but going to hold off on approving until I play around with what a map of `anyFlag` is going to look like in go

**Comment by toddbaert** [comment]
> Looks good to me - but going to hold off on approving until I play around with what a map of `anyFlag` is going to look like in go

:100: , yep, that was my only mild concern too. I thought it seemed like a nice implementation but it'd be good to have that validated. Thanks!

File: protobuf/schema/v1/schema.proto [hunk 4]
```diff
@@ -8,7 +8,39 @@ option csharp_namespace = "OpenFeature.Flagd.Grpc";
...
+    string reason = 1;
+
+    // The variant name of the returned flag value.
+    string variant = 2;
+
+    // The response value of the boolean flag evaluation, will be unset in the case of error.
+    oneof value {
+        bool                    bool_value = 3;
+        string                  string_value = 4;
+        int64                   int_value = 5;
```
Comment (james-milligan) [inline protobuf/schema/v1/schema.proto:39]: I'm in favour of dropping the `int` type from this definition, all values are stored in the evaluator as `float64`, as a result we don't actually know the 'destination' type in the case of bulk evaluation.

Comment (toddbaert) [reply protobuf/schema/v1/schema.proto:39]: That makes sense.

Comment (toddbaert) [reply protobuf/schema/v1/schema.proto]: resolved.

**Review by james-milligan — APPROVED** [review]

**Comment by james-milligan** [comment]
LGTM :)

**Pull request merged**